tests/integration/HACKING.md

```diff
@@ -0,0 +1,468 @@
+
+Currently the Worker plugin supports two ```otherFields```:
+
++ **celeryTaskName** should be the name of the task object. This is almost always the python importable path to the task as a string (unless you have specifically set the [name](http://docs.celeryproject.org/en/latest/userguide/tasks.html#names) when decorating the function). In our example this is the string "my_custom_package.tasks.fibonacci". If you do not specify a ```celeryTaskName``` in ```otherFields``` it will use "girder_worker.run" by default.
++ **celeryQueue** allows you to place the produces message on a different RabbitMQ queue than the default ```celery``` queue.
+
+For the final part of our example we use the jobModel singleton to save the job to the database and then schedule the job to execute. 
+
+### Notes about the Traditional method
+
+In general the traditional method is not the preferred method for producing tasks from Girder. As we will see, it takes several additional steps compared to the "Celery" method. More importantly, the traditional method can ***only*** use the default Celery [configurations](http://docs.celeryproject.org/en/latest/userguide/configuration.html) with the exception of the backend, and broker settings (for why, see "Issues with the Traditional Method" under the architectural notes section). 
+
+The one advantage of the traditional approach is that it does not require the package that defines your task objects to be installed in the Producer's environment (e.g. ```my_custom_package``` in our example does not have to be installed on the same server as Girder). This can be helpful if for some reason you do not have deploy time control over the Girder environment,  or if you have a particularly difficult set of dependencies that would need to be installed in both the Girder and Girder Worker Python environments (though,  See: "What if my task has complex install dependencies?" for suggestions on how to mitigate this issue with the Celery method). 
```

Comment:
In fact you would not need to bring down the girder instances,   only install the package in the girder instance's python environment.  For custom Celery tasks the Worker's will have to be restarted (to pick up the new tasks) - this should be primarily mitigated by the fact that many "user generated" tasks will be run with the generic 'docker run' task